tap_linkedin_ads/streams.py

```diff
@@ -0,0 +1,699 @@
+            'count': page_size,
+            **self.params # adds in endpoint specific, sort, filter params
+        }
+
+        querystring = '&'.join(['%s=%s' % (key, value) for (key, value) in endpoint_params.items()])
+        next_url = 'https://api.linkedin.com/v2/{}?{}'.format(self.path, querystring)
+
+        while next_url: #pylint: disable=too-many-nested-blocks
+            LOGGER.info('URL for %s: %s', self.tap_stream_id, next_url)
+
+            # Get data, API request
+            data = client.get(
+                url=next_url,
+                endpoint=self.tap_stream_id)
+            # time_extracted: datetime when the data was extracted from the API
+            time_extracted = utils.now()
+
+            # Transform data with transform_json from transform.py
+            #  This function converts unix datetimes, de-nests audit fields,
+            #  tranforms URNs to IDs, tranforms/abstracts variably named fields,
+            #  converts camelCase to snake_case for fieldname keys.
+            # For the Linkedin Ads API, 'elements' is always the root data_key for records.
+            # The data_key identifies the collection of records below the <root> element
+            transformed_data = [] # initialize the record list
+            if self.data_key in data:
+                transformed_data = transform_json(data, self.tap_stream_id)[self.data_key]
+            if not transformed_data or transformed_data is None:
+                LOGGER.info('No transformed_data')
+                break # No data results
+
+            pre_singer_transformed_data = copy.deepcopy(transformed_data)
+            if self.tap_stream_id in selected_streams:
+                # Process records and gets the max_bookmark_value and record_count for the set of records
+                max_bookmark_value, record_count = self.process_records(
+                    catalog=catalog,
+                    records=transformed_data,
+                    time_extracted=time_extracted,
+                    bookmark_field=bookmark_field,
+                    max_bookmark_value=max_bookmark_value,
+                    last_datetime=last_datetime,
+                    parent_id=parent_id)
+                LOGGER.info('%s, records processed: %s', self.tap_stream_id, record_count)
+                total_records = total_records + record_count
+
+            # Loop thru parent batch records for each children objects
+            for child_stream_name in children:
+                if child_stream_name in selected_streams:
+                    # For each parent record
+                    child_obj = STREAMS[child_stream_name]()
+
+                    for record in pre_singer_transformed_data:
+
+                        parent_id = record.get(child_obj.foreign_key)
+
+                        child_stream_params = child_obj.params
+                        # Add children filter params based on parent IDs
+                        if self.tap_stream_id == 'accounts':
```

Comment:
Can we remove this if else and move it to specific class?

Reply:
Actually, we have if else condition just for 2 streams. Moving it to a specific class will result in major duplication. Also, [there ](https://github.com/singer-io/tap-linkedin-ads/blob/TDL-20154-dict-base-to-class-base-implementation/tap_linkedin_ads/streams.py#L365) is one condition based on which we want to print a warning message and then continue it. So, I think it would be better to keep the `if else` condition here.